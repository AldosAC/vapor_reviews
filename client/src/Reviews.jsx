import React from 'react';
import Review from './Review.jsx';
import styled from 'styled-components';

const Reviews = ({reviews, sort}) => (
  <ReviewsWrapper> 
    <Title>MOST {sort.toUpperCase()} REVIEWS</Title>
    <Subtitle>IN THE PAST 30 DAYS</Subtitle>
    {
      reviews.map(review => <Review key={review.post_id} review={review}/>)
    }
  </ReviewsWrapper>
);

const Title = styled.div`
  padding-top: 20px;
  padding-left: 5px;
  font-size: 14px;
  padding-right: 5px;
  color: white;
  display: inline;
`;

const Subtitle = styled(Title)`
  color: #56707f;
`;

const ReviewsWrapper = styled.div`
  float: left;
  width: 100%;
  font-family: "Motiva Sans", Arial, Helvetica, sans-serif;
  background: inherit;
  @media only screen and (min-width: 768px) {
    width: auto;
  }
`;

export default Reviews;