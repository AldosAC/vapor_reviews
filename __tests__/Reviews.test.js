import React from 'react';
import { shallow, mount } from 'enzyme';
import Reviews from '../client/src/Reviews.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';
import $ from 'jquery';

Enzyme.configure({ adapter: new Adapter() });

beforeEach(() => jest.resetModules());

describe('Reviews component', () => {
  it('should render elements', () => {
    const wrapper = shallow(
      <Reviews reviews={['test']} sort={'helpful'}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});