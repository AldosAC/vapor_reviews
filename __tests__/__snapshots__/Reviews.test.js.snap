// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component should render elements 1`] = `
<styled.div>
  <styled.div>
    MOST HELPFUL REVIEWS
  </styled.div>
  <Styled(styled.div)>
    IN THE PAST 30 DAYS
  </Styled(styled.div)>
  <Review
    review="test"
  />
</styled.div>
`;
